pkg/types/retry.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package types
+
+import (
+	"math"
+	"time"
+)
```

Comment:
I've used the https://github.com/cenkalti/backoff library before in previous projects and it's pretty small and simple and might reduce the amount of code we would need to keep in ACK itself.

@vijtrip2 would you mind taking a look at whether the [`github.com/cankalti/backoff.ExponentialBackoff:NextBackOff`](https://github.com/cenkalti/backoff/blob/v4/exponential.go#L116) function might be used instead of custom code?

The https://github.com/cenkalti/backoff/ library is MIT-licensed and so not concerning from an attribution perspective.

Reply:
I will definitely take a look at this library and see if I can make use of it here instead of the custom code(which to be honest is very less.)

Reply:
@vijtrip2 actually, look what I found :) https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L380